src/libostree/ostree-repo-commit.c

```diff
@@ -4279,8 +4279,18 @@ import_one_object_direct (OstreeRepo    *dest_repo,
                                            cancellable, error))
                 return FALSE;
             }
+          else if (dest_repo->mode == OSTREE_REPO_MODE_BARE_USER_ONLY)
```

Comment:
That's actually intentional!  It's OK if a bare-user-only repo has a `user.ostreemeta` xattr; it will just be ignored.

The idea is that we want to make it cheap to do the conversion and not require a copy.